### PR TITLE
Correctly identify Skylake CPUs on Darwin.

### LINKED ISSUE #13376: Skylake microarchitecture misidentified on Darwin
On Darwin (Mojave), a Skylake CPU is misidentified as Broadwell.

Upon investigation of the output of `sysctl -n machdep.cpu.{,leaf7_}features`, it would appear that the `xsavec` and `xsaveopt` features are represented by the single flag `XSAVE`, and `clflushopt` is identified as `CLFSOPT`.

### PR DESCRIPTION
**Pull request by chissg**
Fixes #13376 

Uses the already-demonstrated mechanism in `detect.sysctl()` to translate `XSAVE` -> `xsavec xsaveopt` and `CLFSOPT` -> `clflushopt`.

### CODE REVIEW

File: lib/spack/spack/test/llnl/util/cpu.py [hunk 1]
```diff
@@ -156,6 +157,7 @@ def test_architecture_family(target_name, expected_family):
     ('skylake', 'sse3'),
     ('power8', 'altivec'),
     ('broadwell', 'sse4.1'),
+    ('skylake', 'clflushopt'),
```
Comment (alalazo) [inline lib/spack/spack/test/llnl/util/cpu.py]: This is fine to have here, but I want to clarify that it wouldn't have caught the issue you fixed for Mac OS detection. The `info` dictionary above is built for detection and the features in there are best-matched to the target we have in the JSON file. This unit test instead *starts* from one of the targets in the JSON file and checks that it has an expected feature.

Comment (chissg) [reply lib/spack/spack/test/llnl/util/cpu.py]: Fair enough, thanks.

File: lib/spack/llnl/util/cpu/microarchitectures.json [hunk 2]
```diff
@@ -400,7 +400,8 @@
         "adx",
         "clflushopt",
         "xsavec",
-        "xsaveopt"
+        "xsaveopt",
+        "mpx"
```
[the remark this answers is not in the document]
Comment (chissg) [reply lib/spack/llnl/util/cpu/microarchitectures.json]: I'm happy ditching this.

[the remark this answers is not in the document]
Comment (alalazo) [reply lib/spack/llnl/util/cpu/microarchitectures.json]: That's fine with me. If we remove the `mpx` instruction set that's green light from my side.

**Comment by becker33** [comment]
Approved pending removal of `mpx`

**Pull request merged**